device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionManager.java

```diff
@@ -29,7 +30,7 @@
 
     private static final int MAX_WAIT_TO_AUTHENTICATE_MS = 10*1000;
 
-    private final ObjectLock openLinksLock = new ObjectLock();
+    private final CountDownLatch openLinksLock = new CountDownLatch(1);
```

Comment:
what is the reason to change from lock to latch while we are doing 1 as the count down anyways?

Reply:
Timing matters more for locks than it does for latches. If the notifyLock() is called before the lock calls waitLock(), the lock will block and timeout because the notifyLock() is expected to happen after calling waitLock(). With latches, if the countdown() happens before the await(), the latch won't block and time out.